### ISSUE TITLE
Builder initialize before mount / Passing Properties to builder failed

### ISSUE THREAD

**Issue by falconeri** [rappasoft/laravel-livewire-tables#690]
* laravel-livewire-tables Version: 2.0.0
 * laravel Version: 9.6.0

### Description:

I want to pass property to builder with this scenario:
```
public function mount(User $user)
{
   $this->user= $user;
}

public function builder(): Builder
{
    return Store::query()->where('user_id', $this->user->id);
}
```
but `$this->user` is null inside builder, any idea how do i access $user inside builder? thank you

**Comment by falconeri** [comment]
> i also have the same problem

here what i did to solve the problem

```
public $campaign;

public function mount(Campaign $campaign)
{
    $this->campaign = $campaign;
}

public function getRows()
{
    $builder = $this->baseQuery();
    return $this->executeQuery($builder->whereCampaignId($this->campaign->id));
}
```

**Comment by Lootfi** [comment]
Shouldn't the mount method be executed first?

**Comment by falconeri** [comment]
> Shouldn't the mount method be executed first?

The query builder in V2 called in boot, before mount, so it's get executed first before mount

**Comment by gboquizosanchez** [comment]
This can be a possible solution:

```  
public function mount(User $user)
{
    $this->user = $user;
}

public function builder(): Builder
{
    return Store::query()->when(
        $this->user ?? null
        fn ($q) => $q->where('user_id', $this->user->id),
    );
}
```

This worked for me and even I can sort asc or desc properly:

```
public string $category;

public function builder(): Builder
{
    return $this->model::when(
        $this->category ?? null,
        fn ($query) => $query->where('category_id', $this->category),
    );
}
```

I hope this can help you.

Cheers!

**Comment by gid-m** [comment]
How do I use a property in the `public function columns(): array {}` function? For example to hide/show a column using `->hideIf()` ?

**Comment by rappasoft** [comment]
This is a tough one to fix without breaking everyone's current tables. Open to suggestions.

**Comment by rappasoft** [comment]
I need help testing this as i'm not comfortable releasing this big of a change as a minor yet.

If you have time, pull branch `v2-query-rework`, which should fix the issue, but it might have broken other things that I haven't tested (the tests are still a mess).

Here is the commit: https://github.com/rappasoft/laravel-livewire-tables/commit/bce28db3c710c172b4e0e17c0e4102c241512e0d

**Comment by gid-m** [comment]
Thank you @rappasoft. I'll have a look and retort. 

Meanwhile, I found a really hacky way to access the mounted properties. Probably a bad practice but I call `$this->boot()` in the `mount()` function and I'm able to access the properties pretty much anywhere.

**Comment by rappasoft** [comment]
Fixed in v2.4
